src/internal/puck/components/Header.tsx

```diff
@@ -31,18 +30,13 @@ import {
 } from "../ui/Tooltip.tsx";
 import "../../../components/index.css";
 
-const handleClick = (slug: string) => {
-  window.open(`/${slug}`, "_blank");
-};
-
 export const customHeader = (
   handleClearLocalChanges: () => void,
   handleHistoryChange: (histories: History[], index: number) => void,
   data: Data,
   handleSaveData: (data: Data) => Promise<void>,
   isDevMode: boolean
 ) => {
-  const entityDocument = useDocument<any>();
```

Comment:
Is useDocument used anywhere else? If not I think we can remove @yext/pages as a dependency. @mkilpatrick this might solve our double document provider situation. We can just use the DocumentProvider in the starter and not worry about passing the document to the Editor component from ve-lib.

Reply:
That would give us this usage case: 
```tsx
const Edit: () => JSX.Element = () => {
  const entityDocument = useDocumentProvider();

  return (
      <DocumentProvider value={entityDocument}>
        <Editor puckConfigs={puckConfigs} />
      </DocumentProvider>
  );
};
```

Reply:
I can do this in a separate PR @tsaikatie, kinda out of scope for this one, this looks good

Reply:
Tried this and it didn't work 😞 seems that getting rid of the double DocumentProvider won't be quite that simple